### DISCUSSION TITLE
How to abort retry before hitting the retry limit?

### DISCUSSION THREAD

**Discussion by KyleRicardo** [sindresorhus/ky#551] [Q&A] [answered]
Hi community,

I'm using `beforeRetry` hook to do the refresh token logic, and if the refreshToken is expired, I'll receive an error, so I should not try another time and navigate the user to the login page. But even if I return the `ky.stop`, the retry seems not to stop until hit the limit.

What should I do to prevent from keep retrying? Thanks in advance!

**Comment by sholladay** [comment, marked as answer]
We have a test to prove that `ky.stop` works.

https://github.com/sindresorhus/ky/blob/3cf50649ac2a9d5230ebcff94fa58b63a77781dc/test/hooks.ts#L502-L532

It would be helpful to see a reproducible example of what is not working.

Are you sure you are `return`ing the symbol?
